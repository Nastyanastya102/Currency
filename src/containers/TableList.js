import PropTypes from 'prop-types';
//Redux
import { connect } from "react-redux";
//Component
import TableList from '../components/Table/TableList';

const mapStateToProps = ({isLoad: {rates}}) =>({
   rates,
  });

export default connect(
    mapStateToProps,
    null,
  )(TableList);
  
TableList.propTypes = {
    rates: PropTypes.array
};
  