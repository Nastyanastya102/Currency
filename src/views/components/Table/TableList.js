import React from 'react';
import PropTypes from 'prop-types';
//Botstrap
import Table from 'react-bootstrap/Table';

export function TableList({rates}) {
  let tableRow = rates.map( (item, index) => {
        return(
          <tr key={item[0]}>
            <td>{index + 1}</td>
            <td>{item[0]}</td>
            <td>{item[1]}</td>
          </tr>
        );
  });
    return(
   <Table striped bordered hover variant="dark">
          <thead>
            <tr>
              <th>#</th>
              <th>Currency</th>
              <th>Exchange Rate = 1 EUR</th>
            </tr>
          </thead>
          <tbody>         
             {tableRow}
          </tbody>
  </Table>
    )
};


TableList.propTypes = {
  rates: PropTypes.array.isRequired
};
