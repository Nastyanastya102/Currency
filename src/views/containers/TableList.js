import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import TableList from '../components/TableList/TableList';
import { filterList } from '../../redux/ducks/currency/currencyDuck';

const mapStateToProps = ({ currencyApp: { dataFromAPI } }) => ({
  dataFromAPI,
});
const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      filterList,
    },
    dispatch
  );

export default connect(mapStateToProps, mapDispatchToProps)(TableList);
