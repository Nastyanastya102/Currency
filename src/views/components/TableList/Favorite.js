import React from 'react'
import { Star } from './Icon';

import Table from 'react-bootstrap/Table';

const Favorite = ({ gettingData,filterList }) => {

 if(!gettingData.filtered.length) { 
 return(
  <Table striped bordered hover variant="dark">
    <thead>
      <tr>
        <th>Add your favorite currency</th>
      </tr>
    </thead>
  </Table>
)}

let tableRow = gettingData.filtered.map((item, index) => {
    return (
      <tr key={index}>
        <td>{index + 1}</td>
        <td>{item[0]}</td>
        <td>{item[1]}</td>
        <td>
          <i>
            <Star color={true}/>
          </i>
        </td>
      </tr>
    );
  });
  return(
    <Table striped bordered hover variant="dark">
      <thead>
        <tr>
          <th>#</th>
          <th>Currency</th>
          <th>Exchange Rate = 1 EUR</th>
        </tr>
      </thead>
      <tbody>
        {tableRow}
      </tbody>
    </Table>
  )
};
export default Favorite