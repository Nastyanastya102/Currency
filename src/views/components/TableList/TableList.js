import React from 'react';
import PropTypes from 'prop-types';

import Table from 'react-bootstrap/Table';
import { Star } from './Icon';
  
const TableList = ({ dataFromAPI, filterList }) => {

  const list = dataFromAPI.map((key, index) => (
    <tr key={key.key}>
      <td>{index + 1}</td>
        <td>{key.key}</td>
        <td>{key.value}</td>
        <td>
          <i onClick={() => setTimeout(() => filterList(key, index), 0)}>
              <Star color={key.star}/>
          </i>
        </td>
    </tr>
  ));

  return (
    <Table striped bordered hover variant="dark">
      <thead>
        <tr>
          <th>#</th>
          <th>Currency</th>
          <th>Exchange Rate = 1 EUR</th>
        </tr>
      </thead>
      <tbody>
        {list}
      </tbody>
    </Table>
  );
};

export default TableList;

TableList.propTypes = {
  filterList: PropTypes.func.isRequired,
  dataFromAPI: PropTypes.array.isRequired,
};