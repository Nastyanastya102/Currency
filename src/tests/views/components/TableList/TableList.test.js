import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import expect from 'expect';

import TableList from '../../../../views/components/TableList/TableList';

const props = {
  dataFromAPI: [
    { key: 'CAD', value: 1.5118, star: false }, 
    { key: 'HKD', value: 8.4052, star: false }
  ],
  date: "2020-05-08",
  error: false,
  formGridFrome: "EUR",
  formGridTo: "CAD",
  inputText: 0,
  pending: false,
  filterList: jest.fn()
};

describe('Should render MainContent', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<TableList {...props} />);
  });
console.log(wrapper);

  it('renders correctly', () => {
    wrapper = shallow(<TableList {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const headers = wrapper.find('tbody tr');
    expect(headers).toHaveLength(props.dataFromAPI.length);
  });
});
