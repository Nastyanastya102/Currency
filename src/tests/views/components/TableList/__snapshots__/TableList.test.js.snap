// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render MainContent renders correctly 1`] = `
<ForwardRef
  bordered={true}
  hover={true}
  striped={true}
  variant="dark"
>
  <thead>
    <tr>
      <th>
        #
      </th>
      <th>
        Currency
      </th>
      <th>
        Exchange Rate = 1 EUR
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="CAD"
    >
      <td>
        1
      </td>
      <td>
        CAD
      </td>
      <td>
        1.5118
      </td>
      <td>
        <i
          onClick={[Function]}
        >
          <Star
            color={false}
          />
        </i>
      </td>
    </tr>
    <tr
      key="HKD"
    >
      <td>
        2
      </td>
      <td>
        HKD
      </td>
      <td>
        8.4052
      </td>
      <td>
        <i
          onClick={[Function]}
        >
          <Star
            color={false}
          />
        </i>
      </td>
    </tr>
  </tbody>
</ForwardRef>
`;
